test/integration.js

```diff
@@ -0,0 +1,454 @@
+// Native
+const path = require('path')
+const { homedir } = require('os')
+const { URL } = require('url')
+
+// Packages
+const test = require('ava')
+const semVer = require('semver')
+const fkill = require('fkill')
+const { remove, pathExists, readJSON, writeJSON, readFile } = require('fs-extra')
+const execa = require('execa')
+const fetch = require('node-fetch')
+
+// Utilities
+const logo = require('../src/util/output/logo')
+const pkg = require('../package')
+const parseList = require('./helpers/parse-list')
+const removeDeployment = require('./helpers/remove')
+const prepareFixtures = require('./helpers/prepare')
+
+const binary = {
+  darwin: 'now-macos',
+  linux: 'now-linux',
+  win32: 'now-win.exe'
+}[process.platform]
+
+const binaryPath = path.resolve(__dirname, '../packed/' + binary)
+const fixture = name => path.join(__dirname, 'fixtures', 'integration', name)
+const deployHelpMessage = `${logo} now [options] <command | path>`
+const session = Math.random().toString(36).split('.')[1]
+
+// AVA's `t.context` can only be set before the tests,
+// but we want to set it within as well
+const context = {}
+
+const configDir = path.resolve(homedir(), '.now')
+
+const configFiles = {
+  auth: path.resolve(configDir, 'auth.json'),
+  config: path.resolve(configDir, 'config.json')
+}
+
+test.before(async () => {
+  let configContent
+
+  // Close the existing app
+  if (!process.env.CI) {
+    try {
+      await fkill('Now')
+    } catch (err) {}
+  }
+
+  const { auth, config } = configFiles
+
+  // Remove the config directory to
+  // simulate a new user starting the app
+  if (await pathExists(configDir)) {
+    configContent = {
+      auth: await readJSON(auth),
+      config: await readJSON(config)
+    }
+
+    await remove(configDir)
```

Comment:
This seems very dangerous to accidentally run on your computer? We need to either Dockerize this or add a guard that if CI is not set, these tests shouldn't run

Reply:
@rauchg No, because it caches your original config and brings it back.

Reply:
It's still dangerous because the tests can fail and the configuration never be recovered

Reply:
(fail as in _crash_)

Reply:
Removed the local app closing and config restoring.